src/coreclr/jit/simdashwintrinsic.cpp

```diff
@@ -719,6 +719,64 @@ GenTree* Compiler::impSimdAsHWIntrinsicSpecial(NamedIntrinsic       intrinsic,
                     }
                     break;
                 }
+                case NI_VectorT128_Sum:
+                {
+                    if (compOpportunisticallyDependsOn(InstructionSet_SSSE3))
+                    {
+                        GenTree* tmp;
+                        unsigned vectorLength = getSIMDVectorLength(simdSize, simdBaseType);
+                        int      haddCount    = genLog2(vectorLength);
+
+                        for (int i = 0; i < haddCount; i++)
+                        {
+                            op1 = impCloneExpr(op1, &tmp, clsHnd, (unsigned)CHECK_SPILL_ALL,
```

Comment:
Could you post an excerpt from the [dump](https://github.com/dotnet/runtime/blob/main/docs/design/coreclr/jit/viewing-jit-dumps.md) what kind of IR this generates? If I am reading the code correctly, it should be a chain of assignments to `LCL_VAR`s. Is that correct?

Reply:
This is the generated code for `Vector.Sum<int>`.

```IR
STMT00001 (IL 0x000...0x006)
               [000006] -A----------              *  ASG       simd32 (copy)
               [000004] D------N----              +--*  LCL_VAR   simd32<System.Numerics.Vector`1[Int32]> V02 tmp1
               [000003] n-----------              \--*  OBJ       simd32<System.Numerics.Vector`1[Int32]>
               [000002] ------------                 \--*  ADDR      byref
               [000001] -------N----                    \--*  LCL_VAR   simd32<System.Numerics.Vector`1[Int32]> V00 arg0

***** BB02
STMT00002 (IL   ???...  ???)
               [000012] -A----------              *  ASG       simd32 (copy)
               [000010] D------N----              +--*  LCL_VAR   simd32<System.Numerics.Vector`1[Int32]> V03 tmp2
               [000009] ------------              \--*  HWINTRINSIC simd32 int HorizontalAdd
               [000008] ------------                 +--*  LCL_VAR   simd32<System.Numerics.Vector`1[Int32]> V02 tmp1
               [000007] ------------                 \--*  LCL_VAR   simd32<System.Numerics.Vector`1[Int32]> V02 tmp1

***** BB02
STMT00003 (IL   ???...  ???)
               [000018] -A----------              *  ASG       simd32 (copy)
               [000016] D------N----              +--*  LCL_VAR   simd32<System.Numerics.Vector`1[Int32]> V04 tmp3
               [000015] ------------              \--*  HWINTRINSIC simd32 int HorizontalAdd
               [000014] ------------                 +--*  LCL_VAR   simd32<System.Numerics.Vector`1[Int32]> V03 tmp2
               [000013] ------------                 \--*  LCL_VAR   simd32<System.Numerics.Vector`1[Int32]> V03 tmp2

***** BB02
STMT00004 (IL   ???...  ???)
               [000026] -A----------              *  ASG       int
               [000025] D------N----              +--*  LCL_VAR   int    V05 tmp4
               [000024] ------------              \--*  HWINTRINSIC int    int ToScalar
               [000023] ------------                 \--*  HWINTRINSIC simd16 int Add
               [000022] ------------                    +--*  HWINTRINSIC simd16 int ExtractVector128
               [000020] ------------                    |  +--*  LCL_VAR   simd32<System.Numerics.Vector`1[Int32]> V04 tmp3
               [000021] ------------                    |  \--*  CNS_INT   int    1
               [000019] ------------                    \--*  LCL_VAR   simd32<System.Numerics.Vector`1[Int32]> V04 tmp3

***** BB02
STMT00005 (IL 0x006...  ???)
               [000028] ------------              *  RETURN    int
               [000027] ------------              \--*  LCL_VAR   int    V05 tmp4
```

Reply:
Thank you, that looks as expected.

My suggestion would be to use more descriptive names for the various locals that are all `op1` right now, to make it easier to understand the construction, and to move it to (two) dedicated methods. I realize this is not the style this function is using right now, but it is being rewritten to do that in #53450.

Reply:
I can do that. I wanted to keep the style of the original file as much as possible with accordance to PR guidelines. @tannergooding what do you think?

Reply:
I don't have a strong preference either way. The file is eventually (once 53450 is reviewed/approved) going to move towards having something like a  `gtNewSimdSumNode`, but I can also do that as part of 53450 or a later PR.

Reply:
I see that you want this in compiler.h/cpp. I would prefer to do this in a separate PR after your work on 53450 is completed, so I can study it carefully beforehand. I would love to try and help more in the JIT, but for the moment some of the concepts here are too new/fresh/unknown to me :-) If you can point me to any beginner/intermediate issues, would be awesome.

Reply:
Sounds good. I'll likely end up doing the small fixup required as part of my follow up PR to 53450, since there are a couple more xplat methods I'll need to implement anways :)